### PR TITLE
Use tx cost endpoint and display fee in confirmation dialog

### PR DESCRIPTION
**Pull request by liviavinci**

### CODE REVIEW

**Comment by liviavinci** [comment]
screen shot 2018-05-17 at 3 39 10 pm
screen shot 2018-05-17 at 3 39 23 pm
screen shot 2018-05-17 at 3 39 37 pm
screen shot 2018-05-17 at 3 39 49 pm
screen shot 2018-05-17 at 3 40 02 pm

File: src/components/TxConfirmDialog/index.js [hunk 1]
```diff
@@ -2,44 +2,104 @@ import React, { Component } from 'react';
...
+    const txFee = _.sumBy(transactionCost, (cost) => cost.gasCost ? parseFloat(cost.gasCost) : 0);
 
     return (
       <Dialog open={isOpen}>
         <DialogTitle>
           <FormattedMessage id="txConfirm.title" defaultMessage="Please Confirm Your Transaction" />
         </DialogTitle>
         <DialogContent>
-          {formatMessage(messages.confirmMessage, { txDesc, txAmount, txToken })}
+          { txFee ? formatMessage(messages.confirmMessageWithFee, { txDesc, txAmount, txToken, txFee }) : formatMessage(messages.confirmMessage, { txDesc, txAmount, txToken }) }
```
Comment (alex-cory) [inline src/components/TxConfirmDialog/index.js]: no space. i.e.
`{txFee ? `
please see [React Best Practices in frontify](https://bodhi.frontify.com/document/230177#/react-best-practices)

File: src/components/TxConfirmDialog/index.js [hunk 2]
```diff
@@ -2,44 +2,104 @@ import React, { Component } from 'react';
...
+          { transactionCost.length ?
```
Comment (alex-cory) [inline src/components/TxConfirmDialog/index.js]: 1. same as above
2. please put parenthesis on same line
3. please don't use ternaries in places like this
`{!!transactionCost.length && (`
please see [React Best Practices in frontify](https://bodhi.frontify.com/document/230177#/react-best-practices)

File: src/components/TxConfirmDialog/index.js [hunk 3]
```diff
@@ -2,44 +2,104 @@ import React, { Component } from 'react';
...
 @injectIntl
+@withStyles(styles, { withTheme: true })
 @connect((state) => ({
   txConfirmInfoAndCallback: state.App.get('txConfirmInfoAndCallback'),
+  transactionCost: state.App.get('transactionCost'),
 }), (dispatch) => ({
   clearTxConfirm: () => dispatch(appActions.clearTxConfirm()),
+  getTransactionCost: (txInfo) => dispatch(appActions.getTransactionCost(txInfo)),
 }))
+
```
Comment (alex-cory) [inline src/components/TxConfirmDialog/index.js]: please keep these together.  The decorator is a function wrapping the component. We don't call functions like this do we?
```js
withStyles(styles, { withTheme: true })(
  TxConfirmDialog
);
```
no we do it like
```js
withStyles(styles, { withTheme: true })(TxConfirmDialog);
```
so let's keep them together please :)
```js
@withStyles(...)
export default ...
```

File: src/components/TxConfirmDialog/index.js [hunk 4]
```diff
@@ -2,44 +2,104 @@ import React, { Component } from 'react';
...
           <FormattedMessage id="txConfirm.title" defaultMessage="Please Confirm Your Transaction" />
         </DialogTitle>
         <DialogContent>
-          {formatMessage(messages.confirmMessage, { txDesc, txAmount, txToken })}
+          {txFee ? formatMessage(messages.confirmMessageWithFee, { txDesc, txAmount, txToken, txFee }) : formatMessage(messages.confirmMessage, { txDesc, txAmount, txToken })}
+          {Boolean(transactionCost.length) && (
+            <Table className={classes.costTable}>
+              <TableHead>
+                <TableRow>
+                  <TableCell>
```
Comment (alex-cory) [inline src/components/TxConfirmDialog/index.js]: Can we make a `Cell` component? i.e.
```js
const Cell = injectIntl(({ id, message, intl }) => (
  <TableCell>
    {intl.formatMessage({ id, message })}
  </TableCell>
))
```

**Pull request merged**